src/librustc/hir/map/def_collector.rs

```diff
@@ -27,7 +27,7 @@ pub struct DefCollector<'a> {
     hir_crate: Option<&'a hir::Crate>,
     definitions: &'a mut Definitions,
     parent_def: Option<DefIndex>,
-    pub visit_macro_invoc: Option<&'a mut FnMut(NodeId, DefIndex)>,
+    pub visit_macro_invoc: Option<&'a mut FnMut(NodeId, DefIndex, bool /* const_integer */)>,
```

Comment:
I think a closure is as good as a trait object, but probably better to take a struct rather than a 3-tuple.